sdks/xmtp-cli/commands/groups.ts

```diff
@@ -0,0 +1,245 @@
+        config.targetAddress as `0x${string}`,
+      );
+
+      console.log(`✅ DM created: ${conversation.id}`);
+      console.log(`🔗 URL: https://xmtp.chat/conversations/${conversation.id}`);
+    } else {
+      // group
+      if (
+        (!config.memberAddresses || config.memberAddresses.length === 0) &&
+        (!config.memberInboxIds || config.memberInboxIds.length === 0)
+      ) {
+        console.error(
+          `❌ At least one of --member-addresses or --member-inbox-ids is required when creating a group`,
+        );
+        process.exit(1);
+      }
+
+      const groupName = config.groupName || `Group ${Date.now()}`;
+      const groupDescription =
+        config.groupDescription || "Group created by XMTP groups CLI";
+
+      console.log(`🚀 Creating empty group...`);
+
+      // Create an empty group first - use createGroupWithAddresses with empty array
+      const group = await agent.createGroupWithAddresses(
+        [] as `0x${string}`[],
+        {
+          groupName,
+          groupDescription,
+        },
+      );
+
+      // Add members by addresses
+      if (config.memberAddresses && config.memberAddresses.length > 0) {
+        console.log(
+          `Adding ${config.memberAddresses.length} members by address...`,
+        );
+        await group.addMembersByIdentifiers(
+          config.memberAddresses.map((member) => ({
+            identifier: member,
+            identifierKind: IdentifierKind.Ethereum,
+          })),
+        );
+      }
+
+      // Add members by inbox IDs
+      if (config.memberInboxIds && config.memberInboxIds.length > 0) {
+        console.log(
+          `Adding ${config.memberInboxIds.length} members by inbox ID...`,
+        );
+        await group.addMembers(config.memberInboxIds);
+      }
+
+      console.log(`✅ Group created: ${group.id}`);
+      console.log(`📝 Name: ${group.name}`);
+      console.log(`🔗 URL: https://xmtp.chat/conversations/${group.id}`);
+    }
+  } catch (error) {
+    console.error(
+      `❌ Failed: ${error instanceof Error ? error.message : String(error)}`,
+    );
+    process.exit(1);
+  }
+}
+
+async function runMetadataOperation(config: {
+  groupId?: string;
+  groupName?: string;
+  groupDescription?: string;
+  imageUrl?: string;
+}): Promise<void> {
+  if (!config.groupId) {
+    console.error(`❌ --group-id is required`);
+    process.exit(1);
+  }
+
+  if (!config.groupName && !config.groupDescription && !config.imageUrl) {
+    console.error(
+      `❌ At least one of --name, --description, or --image-url is required`,
+    );
+    process.exit(1);
+  }
+
+  console.log(`🔄 Updating group metadata: ${config.groupId}`);
+
+  const agent = await getAgent();
+
+  try {
+    const conversation = await agent.client.conversations.getConversationById(
+      config.groupId,
+    );
+
+    if (!conversation) {
+      console.error(`❌ Group not found: ${config.groupId}`);
+      process.exit(1);
+    }
+
+    const group = conversation as Group;
```

Comment:
it would be best to check if `conversation` is an instance of `Group` instead. if not, exit early.